libraries/SuplaDevice/supla/network/esp_wifi.h

```diff
@@ -68,24 +74,60 @@ class ESPWifi : public Supla::Network {
     }
     Serial.println("]");
 #endif
-    long sendSize = client.write((const uint8_t *)buf, count);
+    long sendSize = client->write((const uint8_t *)buf, count);
     return sendSize;
   }
 
   bool connect(const char *server, int port) {
-    return client.connect(server, port);
+    String message;
+    if (client == NULL) {
+      if (isSecured) {
+        message = "secured connection";
+        client = new WiFiClientSecure();
+        if (fingerprint.length() > 0) {
+          message += " with certificate matching";
+          ((WiFiClientSecure *)client)->setFingerprint(fingerprint.c_str());
+        } else {
+          message += " without certificate matching";
+          ((WiFiClientSecure *)client)->setInsecure();
```

Comment:
If needed, use C++ casting style.
Here, you can create temperaru variable of WifiClientSecure, execute all opeations on it and then assign to "client" pointer. Then casting will not be needed